src/gui/dialogs/campaign_selection.cpp

```diff
@@ -30,15 +29,119 @@
 #include "gui/widgets/tree_view_node.hpp"
 #include "gui/widgets/window.hpp"
 #include "preferences/preferences.hpp"
+#include "serialization/markup.hpp"
+#include "utils/irdya_datetime.hpp"
 
 #include <functional>
-#include "utils/irdya_datetime.hpp"
 
 namespace gui2::dialogs
 {
 
 REGISTER_DIALOG(campaign_selection)
 
+void campaign_selection::update_campaign_list()
```

Comment:
Could you merge this back into `pre_show` since it's only used once now? Makes the diff cleaner.

Reply:
Updated (and accidentally closed). See https://github.com/wesnoth/wesnoth/pull/9489.